processor/batchprocessor/metadata.yaml

```diff
@@ -46,3 +46,36 @@ telemetry:
       sum:
         value_type: int
         async: true
+
+config:
+  type: object
+  additionalProperties: false
+  description: "Configuration parameters for the batch processor."
+  properties:
+    timeout:
+      type: string
+      format: duration
+      # TODO: The default duration should be "P200MS"
+      default: "200ms"
+      description: 
+    send_batch_size:
+      type: integer
+      minimum: 0
+      default: 8192
+      description: 
+    send_batch_max_size:
+      type: integer
+      default: 0
+      minimum: 0
```

Comment:
At the moment, `go-jsonschema`'s `main` branch doesn't do anything when it sees constraints such as `minimum` and `maximum`. However, I patched my own fork of `go-jsonschema` so that it supports it. You can see in the `config.go` that there are checks to make sure some integers are not negative.

Reply:
Maybe we can keep uint32 and we won't have to set such constraints?

Reply:
Json schema itself doesn't seem to have an unsigned integer type. It only has `integer` and `number` (float) types, which can have [various constraints](https://json-schema.org/understanding-json-schema/reference/numeric#range).

Reply:
Hello @atoulme @ptodev , go-jsonschema maintainer here. I have just merged a pr that adds support for numeric validation (multipleOf, maximum, exclusiveMaximum, minimum, exclusiveMinimum) and string patterns in main. But there is still no support for uint types. I am open to discuss it if it is needed, maybe we can fit it in with a bit of work.

Reply:
that's neat and appreciated.

Reply:
Hi @omissis, thank you so much for taking a look! I just realised that go-jsonschema already has a `goJSONSchema` property which allows users to override the type. I think it should be sufficient for our purposes.

Hi @atoulme! I updated the PR to use things like squashing structs, using uint, and using opaque string types. You can do it with:
```
goJSONSchema:
  type: uint32
```

And:

```go
goJSONSchema:
  type: configopaque.String
  imports:
  - go.opentelemetry.io/collector/config/configopaque
```

There are examples in the metadata.yaml files.

Are you happy with the way the uints and opaque strings are set up? If you are, I'll resolve this GitHub discussion.